devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -1366,115 +1507,135 @@ void YarpRobotLoggerDevice::run()
     {
         if (m_robotSensorBridge->getIMUMeasurement(sensorName, m_analogSensorBuffer))
         {
-            // it will return a tuple containing the Accelerometer, the gyro and the orientatio
+            // it will return a tuple containing the Accelerometer, the gyro and the orientation
             this->unpackIMU(m_analogSensorBuffer,
                             m_acceloremeterBuffer,
                             m_gyroBuffer,
                             m_orientationBuffer);
 
-            m_bufferManager.push_back(m_acceloremeterBuffer, time, "accelerometers::" + sensorName);
-            m_bufferManager.push_back(m_gyroBuffer, time, "gyros::" + sensorName);
-            m_bufferManager.push_back(m_orientationBuffer, time, "orientations::" + sensorName);
+            signalName = accelerometersName + treeDelim + sensorName;
+            logData(signalName, m_acceloremeterBuffer, time);
+
+            signalName = gyrosName + treeDelim + sensorName;
+            logData(signalName, m_gyroBuffer, time);
+
+            signalName = orientationsName + treeDelim + sensorName;
+            logData(signalName, m_orientationBuffer, time);
         }
     }
 
-    for (const auto& sensorName : m_robotSensorBridge->getCartesianWrenchesList())
+    if (m_streamCartesianWrenches)
     {
-        if (m_robotSensorBridge->getCartesianWrench(sensorName, m_ftBuffer))
+        for (const auto& cartesianWrenchName : m_robotSensorBridge->getCartesianWrenchesList())
         {
-            m_bufferManager.push_back(m_ftBuffer, time, "cartesian_wrenches::" + sensorName);
+            if (m_robotSensorBridge->getCartesianWrench(cartesianWrenchName, m_ftBuffer))
+            {
+                signalName = cartesianWrenchesName + treeDelim
+                             + cartesianWrenchName;
+                logData(signalName, m_ftBuffer, time);
+            }
         }
     }
 
-    std::string signalFullName;
     for (auto& [name, signal] : m_vectorsCollectionSignals)
     {
         std::lock_guard<std::mutex> lock(signal.mutex);
-        const BipedalLocomotion::YarpUtilities::VectorsCollection* collection
+        BipedalLocomotion::YarpUtilities::VectorsCollection* externalSignalCollection
             = signal.client.readData(false);
-        if (collection != nullptr)
+
+        if (externalSignalCollection != nullptr)
         {
-            if (!signal.dataArrived)
+            if(!signal.dataArrived)
```

Comment:
```suggestion
            if (!signal.dataArrived)
```

Reply:
Fixed in [f8d16c4](https://github.com/ami-iit/bipedal-locomotion-framework/commit/f8d16c43c6280982676ab954157a4d7514996c0f) and [69fa835](https://github.com/ami-iit/bipedal-locomotion-framework/commit/69fa835bc1c7725293775adf108c6e5fd53aec7e)